### ISSUE TITLE
Reduce background memory from navigation state

### ISSUE THREAD

**Issue by noproto** [flipperdevices/flipperzero-firmware#3941]
### Describe the enhancement you're suggesting.

When an app is launched from the main menu, or the file browser (versus the quick launch shortcuts, or `ufbt launch`) it consumes up to 10.5 KB more RAM. 10500 bytes seems excessive to store what amounts to the user navigation state, and it starves memory intense applications since it represents 8% of the available free memory.

If these are suspended threads, perhaps the smallest representative version of their state needs to be cached instead of the entire application remaining in residual memory.

**Comment by skotopes** [comment]
@noproto yep, we know that main menu consumes memory when you launch apps from it. Same goes for archive and couple other places. 
We also were experimenting with serializing state and some other ways to reduce system memory foot print. Sadly all of them affect User Experience. So we decided to leave current state of things.
